integration-tests/test_connection.py

```diff
@@ -6,12 +6,43 @@
 :upstream: Yes
 """
 
+import configparser
+import os
+import typing
+
 import pytest
 
+if typing.TYPE_CHECKING:
+    import pytest_client_tools.insights_client
+
 # the tests need a valid connection to insights so therefore the subman registration
 pytestmark = pytest.mark.usefixtures("register_subman")
 
 
+def _is_using_proxy(
+    insights_config: "pytest_client_tools.insights_client.InsightsClientConfig",
+) -> bool:
+    for key in os.environ.keys():
+        if key.lower() == "https_proxy":
+            return True
+
+    try:
+        config_proxy = insights_config.proxy
+        if config_proxy != "":
+            return True
+    except KeyError:
+        return False
+
+    # sub-man fixture doesn't currently support reading the configuration file,
+    # let's parse it ourselves.
+    rhsm_config = configparser.ConfigParser()
+    rhsm_config.read("/etc/rhsm/rhsm.conf")
+    if rhsm_config.get("server", "proxy_hostname", fallback=None) is not None:
+        return True
```

Comment:
even if it's a very narrow corner case, I'd catch all the exceptions of `ConfigParser` and ignore them, to avoid stacktracing on an invalid `rhsm.conf` or in case the `server` section or the `proxy_hostname` key are missing